### ISSUE TITLE
Order nwp channels

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/ocf_datapipes#326]
## Detailed Description
Should we add a pipeline that orders the nwp channels alphabetically

## Context
- the ml models do carry about what order the channels comes in
- Currently the model might break if the channels are different order. For example if the order was different from what it was trained on

## Possible Implementation
- add bool to config
- add one data pipe that does the ordering
- add into data pipeline 
- (add test)

**Comment by AUdaltsova** [comment]
I looked at where channel selection happens and I **think** this can be achieved via a one-liner in `ocf_datapipes/select/filter_channels.py` (just sort the channel list before performing selection, it should return it in the right order after that, including coord reordering)

**Comment by peterdudfield** [comment]
Thats a good idea

**Comment by peterdudfield** [comment]
It would be interested to know if if `sel` just selected the channels, or it selects them and orders them https://github.com/openclimatefix/ocf_datapipes/blob/main/ocf_datapipes/select/filter_channels.py#L49

by trying `d.sel({"variable": ["lcc", "mcc"]})` and `d.sel({"variable": ["mcc", "lcc"]})` we do seem to get different results

**Comment by peterdudfield** [comment]
Looking at ECMWF, we can see that all values are 0 in the second step

**Comment by AUdaltsova** [comment]
Yes, that's what I was basing the one-liner suggestion on: `sel` seems to reorder coordinates, so the order of channels depends on the order in which somebody adds them into the config, and hence is very prone to inconsistencies.

I am trying to see if I can find a solid description of the reordering somewhere in the docs

**Comment by dfulu** [comment]
I'm not sure I follow what the issue is here. If using the [filter channels function](https://github.com/openclimatefix/ocf_datapipes/blob/main/ocf_datapipes/select/filter_channels.py#L49) which we use [for example here](https://github.com/openclimatefix/ocf_datapipes/blob/main/ocf_datapipes/training/common.py#L162) then we will have the same channel ordering in training and production, even if the dataset we are selecting from has them ina. different order on disk. So long as the input data config remains the same at training and production. I think this is the desired behaviour?

**Comment by dfulu** [comment]
> It would be interested to know if if `sel` just selected the channels, or it selects them and orders them https://github.com/openclimatefix/ocf_datapipes/blob/main/ocf_datapipes/select/filter_channels.py#L49
> 
> by trying `d.sel({"variable": ["lcc", "mcc"]})` and `d.sel({"variable": ["mcc", "lcc"]})` we do seem to get different results

Yes, this is how the selection works. It reorders the channels based on list. The first one has the channels in order lcc then mcc. The second has them in the order mcc then lcc

**Comment by AUdaltsova** [comment]
@dfulu yeah as long as the same config file is used it should be completely fine
